src/lib/autodiscovery/autodiscovery_manager.py

```diff
@@ -1,76 +1,148 @@
             )
-        )
-        self.time_since_last_autodiscovery = datetime.now()
-
-    async def _refresh_autodiscovery_task(self, services, new_extension_versions_hash):
-        if self.autodiscovery_task is not None:
-            result = await self.autodiscovery_task
-            self.autodiscovered_metrics_cache = result.enriched_services
-            self.last_autodiscovered_metric_list_names = result.discovered_metric_list
-        self.autodiscovery_task = asyncio.create_task(
-            enrich_services_with_autodiscovery_metrics(
-                services, self.last_autodiscovered_metric_list_names
+            self.autodiscovery_enabled = False
+
+    @staticmethod
+    def generate_resource_mapping(mapping_yaml: Dict[str, Any]) -> Dict[str, List[ServiceStub]]:
+        resource_mapping = {}
+        for extension in mapping_yaml["gcp_extension_mapping"]:
+            for service in extension["services"]:
+                for monitored_resource in service.get("monitored_resources", []):
+                    resource_name = monitored_resource.get("value")
+                    if resource_name is not None:
+                        resource_mapping.setdefault(resource_name, []).append(
+                            ServiceStub(
+                                extension_name=extension["extension_name"],
+                                service_name=service["service_name"],
+                                feature_set_name=service["feature_set"],
+                            )
+                        )
+        return {
+            resource_name: list(set(service_stub))
+            for resource_name, service_stub in resource_mapping.items()
+        }
+
+    async def check_resources_to_autodiscovery(
+        self, services: List[GCPService]
+    ) -> Dict[str, AutodiscoveryResourceLinking]:
+        prepared_resources = {}
+
+        enabled_services_identifiers = {}
+        disabled_services_identifiers = {}
+
+        for service in services:
+            # com.dynatrace.extension.<%=extensionName%>
+            extension_name = service.extension_name.split(".")[-1]
+            identifier = ServiceStub(extension_name, service.name, service.feature_set)
+            if service.is_enabled:
+                enabled_services_identifiers[identifier] = service
+            else:
+                disabled_services_identifiers[identifier] = service
+
+        for resource in self.autodiscovery_config["searched_resources"]:
+            if resource in self.autodiscovery_resource_mapping:
+                possible_linking = []
+                disabled_linking = []
+
+                for required_service in self.autodiscovery_resource_mapping[resource]:
+                    if required_service in enabled_services_identifiers:
+                        possible_linking.append(enabled_services_identifiers[required_service])
+                    if required_service in disabled_services_identifiers:
```

Comment:
Consider reducing noise in the logic too. E.g., from line 90, we don't need to ask again in 101, we know it will be disabled if it's not enabled (unless I am missing something). Stuff like that.